### PR TITLE
RandomItemPool Fixes

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [New] Random Item Spawners can now be hand activated to manually trigger their behavior if they're disabled by mappers/designers.
CL: [Fix] Fixed an issue where some random-item-spawners would break randomly and refuse to give out loot.
CL: [Fix] Fixed an issue where spawned items from random-item-spawners would refuse to add their content to nearby storage containers.
CL: [Fix] Fixed an issue where random-item-spawners would not delete themselves after spawning their loot.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs [hunk 1]
```diff
@@ -39,68 +42,58 @@ public void OnSpawnServer(SpawnInfo info)
...
-					continue;
-				}
-
-				GenerateItem(pool, UnrestrictedAndspawn);
-				if (UnrestrictedAndspawn == false) return;
+				PoolData pool = GrabRandomPool();
+				if (pool == null) continue;
+				GenerateItem(pool, unrestrictedAndSpawn);
+				spawnAmount++;
+				if (unrestrictedAndSpawn == false) break; // what the fuck does UnrestrictedAndspawn mean? What's the point? terrible name
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:65]: I don't remember doing this unrestricted thing, possibly was Bod trying to fix a bug?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:65]: I'm not sure what it is trying to fix or do, so I'm leaving it as is for the time being.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:65]: https://github.com/unitystation/unitystation/pull/7492

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:65]: I'll leave it as is for now

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**